routes/users.js

```diff
@@ -19,60 +19,24 @@ const validateUser = (req, res, next) => {
     } else {
         next();
     }
-}
-
-// Create an array to store user data
-let users = [];
+};
 
 // Find all users
-router.get('/users', (req, res) => {
-    res.json(users)
-});
+router.get('/users', userController.getUsers);
 
 // Create users
-router.post('/users', validateUser, (req, res) => {
-    const name = req.body.name;
-    const email = req.body.email;
-    const user = {
-        id: uuidv4(),
-        email,
-        name,
-    };
-    users.push(user);
-    res.status(200).json(user);
-});
+router.post('/users', validateUser, userController.createUser);
```

Comment:
The same way you created the `controllers` folder and the `userController.js` file, I want you to create a `middlewares` folder and a `userMiddlewares.js` file and place this `validateUser` middleware in there.

Reply:
i was going to ask about where to place the validation, that got me thinking awhile